python/rascaline/rascaline/utils/clebsch_gordan/_clebsch_gordan.py

```diff
@@ -34,79 +40,100 @@ def _standardize_keys(tensor: TensorMap) -> TensorMap:
     if "species_neighbor" in tensor.keys.names:
         tensor = tensor.keys_to_properties(keys_to_move="species_neighbor")
     keys = tensor.keys.insert(
-        name="order_nu",
-        values=_dispatch.int_array_like([1], like=tensor.keys.values),
         index=0,
+        name="order_nu",
+        values=_dispatch.int_array_like(
+            len(tensor.keys.values) * [1], like=tensor.keys.values
+        ),
     )
     keys = keys.insert(
-        name="inversion_sigma",
-        values=_dispatch.int_array_like([1], like=tensor.keys.values),
         index=1,
+        name="inversion_sigma",
+        values=_dispatch.int_array_like(
+            len(tensor.keys.values) * [1], like=tensor.keys.values
+        ),
     )
     return TensorMap(keys=keys, blocks=[b.copy() for b in tensor.blocks()])
 
 
 def _parse_selected_keys(
     n_iterations: int,
+    array_like: Array,
     angular_cutoff: Optional[int] = None,
-    selected_keys: Optional[Union[Labels, List[Labels]]] = None,
-    like=None,
+    selected_keys: Optional[Union[Labels, List[Union[Labels, None]]]] = None,
 ) -> List[Union[None, Labels]]:
     """
     Parses the `selected_keys` argument passed to public functions. Checks the
     values and returns a :py:class:`list` of :py:class:`Labels` objects, one for
-    each iteration of CG combination.
-
-    `like` is required if a new :py:class:`Labels` object is to be created by
-    :py:mod:`_dispatch`.
+    each iteration of CG combination.The `:param array_like:` determines the
+    array backend of the Labels created
     """
+    # Check the selected_keys
+    if (
+        (selected_keys is not None)
+        and (not isinstance(selected_keys, list))
+        and (not is_labels(selected_keys))
+    ):
+        raise TypeError(
+            "`selected_keys` must be `None`, `Labels` or List[Union[None, `Labels`]]"
+        )
+
+    if isinstance(selected_keys, list):
+        if not all(
+            [
+                is_labels(selected_keys[i]) or (selected_keys[i] is None)
+                for i in range(len(selected_keys))
+            ]
+        ):
+            raise TypeError(
+                "`selected_keys` must be a Labels or List[Union[None, Labels]]"
+            )
+
     # Check angular_cutoff arg
     if angular_cutoff is not None:
         if not isinstance(angular_cutoff, int):
             raise TypeError("`angular_cutoff` must be passed as an int")
         if angular_cutoff < 1:
             raise ValueError("`angular_cutoff` must be >= 1")
 
+    # we use a new variable for selected_keys so TorchScript can infer correct type
+    selected_keys_: List[Union[None, Labels]] = []
```

Comment:
I nevertheless added this new variable, because I find this clearer when tracking the type of a variable. Also type checker probably would have suggested such behavior